### PR TITLE
added a new key extractor to overcome an issue with flatlist

### PR DESCRIPTION
**Pull request by eladgel**
Added a new property

### CODE REVIEW

File: src/agenda/reservation-list/index.tsx [hunk 1]
```diff
@@ -258,7 +260,9 @@ class ReservationList extends Component<ReservationListProps, State> {
     );
   };
 
-  keyExtractor = (_item: DayAgenda, index: number) => String(index);
+  keyExtractor = (_item: DayAgenda, index: number) => {
+      return this.props.reservationKeyExtractor && this.props.reservationKeyExtractor(_item, index) || String(index);
```
Comment (Inbal-Tish) [inline src/agenda/reservation-list/index.tsx:264]: You're not fixing the default key...

**Comment by Inbal-Tish** [comment]
@eladgel Thank you for the pr. There is a lint error that fails the build and some comments. I opened a fix to move things faster - #1795. Hope that will do the trick. Thank you and looking forward to your collaboration again!

File: src/agenda/reservation-list/index.tsx [hunk 2]
```diff
@@ -44,6 +44,8 @@ export type ReservationListProps = ReservationProps & {
   refreshing?: boolean;
   /** If provided, a standard RefreshControl will be added for "Pull to Refresh" functionality. Make sure to also set the refreshing prop correctly */
   onRefresh?: () => void;
+
+  reservationKeyExtractor: PropTypes.Requireable<(...args: any[]) => any>,
```
Comment (Inbal-Tish) [inline src/agenda/reservation-list/index.tsx:48]: No need for such an elaborate name, `keyExtractor` is good enough.

**Comment by eladgel** [comment]
I thought that keyExtractor was already declared and used by Calendar List.
That's why I didn't use it again

**Pull request closed without merging**